pac4j-jwt/src/main/java/org/pac4j/jwt/credentials/authenticator/JwtAuthenticator.java

```diff
@@ -61,14 +60,22 @@ public void validate(TokenCredentials credentials) {
         SignedJWT signedJWT = null;
 
         try {
-            // Parse the JWE string
-            JWEObject jweObject = JWEObject.parse(token);
-
-            // Decrypt with shared key
-            jweObject.decrypt(new DirectDecrypter(this.secret.getBytes("UTF-8")));
-
-            // Extract payload
-            signedJWT = jweObject.getPayload().toSignedJWT();
+            JWT jwt = JWTParser.parse(token);
+
+            if (jwt instanceof SignedJWT) {
+                signedJWT = (SignedJWT) jwt;
+            } else if (jwt instanceof EncryptedJWT) {
+                // Parse the JWE string
+                JWEObject jweObject = JWEObject.parse(token);
+
+                // Decrypt with shared key
+                jweObject.decrypt(new DirectDecrypter(this.secret.getBytes("UTF-8")));
```

Comment:
@mmoayyed : what did you expect? A `encoding` property which can be set? What would be the use case not to use `UTF-8`?